src/ticks/debugger.c

```diff
@@ -1733,3 +1747,38 @@ static void print_hotspots()
         fclose(fp);
     }
 }
+
+void stdout_log(const char *fmt, ...)
+{
+    va_list args;
+    int len;
+
+    /* Initialize a variable argument list */
+    va_start(args, fmt);
+
+    /* Get length of format including arguments */
+    len = vsnprintf(NULL, 0, fmt, args);
+
+    /* End using variable argument list */
+    va_end(args);
+
+    if (len < 0) {
+        /* vsnprintf failed */
+        return;
+    } else {
+        /* Declare a character buffer for the formatted string */
+        char formatted[len + 1];
+
+        /* Initialize a variable argument list */
```

Comment:
I need to check another project but I’ve got a memory of one platform not liking the reuse of a va_list - a va_copy was needed.

Reply:
Haha this was stolen fresh from the internet, if you've got a better alternative feel free to suggest :D 

The goal is to basically have two implementations of printf(fmt, ...), because with mi/2 you can't just printf, there's syntax to follow

Reply:
You need something like this:

```
// Copes with some versions of visual studio
#ifndef va_copy
#define va_copy(a,b) (a)=(b)
#endif


....

void stdout_log(const char *fmt, ...)
 {
     va_list args;
     va_list args2;
     int len;

     /* Initialize a variable argument list */
     va_start(args, fmt);
     va_copy(args2,args)

     /* Get length of format including arguments */
     len = vsnprintf(NULL, 0, fmt, args2);

     /* End using variable argument list */
     va_end(arg2);
...
Rest of code as it was
```

Reply:
Ok, I made the change. Works on my MAC. Something like this?